### PR TITLE
Added code saving and next button.

### PR DESCRIPTION
**Pull request by Nick-McGee**
- User's code is now saved to the JSON file.
- 'nextLocked' is saved to JSON now. This will let the program remember if a user passed those lessons or not. This still has a bug where if you press 'back' and then 'next' on the last part of a lesson then 'next' is unlocked on the last part. This should not happen as there is no 'next' past the last part.

- TODO: Code reset. Reverts the user's code back to how it was if they feel like are too far down the wrong path.
- TODO: Lesson reset. Resets the entire lesson for a new user.
- TODO: Clean up main.js. It's a bit of a mess now but I will refactor it a bit during reading week.

### CODE REVIEW

**Comment by Ansper11** [comment]
Hi Nick, I'm a little confused why we don't save the lesson to HTML file as we did in the last term? Or you mean we will save users' current states in a JSON file?

File: webserver/static/js/main.js [hunk 1]
```diff
@@ -61,8 +64,32 @@ document.getElementById("homepage").addEventListener('click', function() {
 document.getElementById("run").addEventListener('click', async function() {
     playBtnClick();
 
+    // This should be turned into a function later
+    // Lesson data might be turned into a global variable
+    // Save the code
+    switch(currentLesson) {
+        case 1:
+            lessonData = await getJSONData(1);
+            lessonData[lesson1Part].code = editor.getValue()
```
Comment (Ansper11) [inline webserver/static/js/main.js:73]: It's this means the user can save their code for later?

Comment (Nick-McGee) [reply webserver/static/js/main.js:73]: Yes, code will be saved to the JSON file there.

**Comment by Nick-McGee** [comment]
> Hi Nick, I'm a little confused why we don't save the lesson to HTML file as we did in the last term? Or you mean we will save users' current states in a JSON file?

Minecraft Education only allows one HTML file at a time. By using JSON, we can save a lot of code by writing just one template and filling it with the JSON data. Additionally, we can store other data with the JSON, such as saving their work, or the original code in case a user wants to reset their code

**Comment by eloiseespel** [comment]
Not much to say the code is great, your effort to add in comments is appreciated. I think that moving the Json is awesome, thanks for working on that.

**Pull request merged**